src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -144,12 +165,33 @@ public Object invokeMethod(String name, Object args) {
                     throw new GroovyRuntimeException(x);
                 }
             }
+            Set<String> ambiguousFunctionNames = new HashSet<>(0);
             for (StepDescriptor d : StepDescriptor.all()) {
-                functions.put(d.getFunctionName(), d);
+                String functionName = d.getFunctionName();
+                if (functions.containsKey(functionName)) {
+                    ambiguousFunctionNames.add(functionName);
+                }
+                // TODO: Switch we switch to putIfAbsent so that the descriptor with the highest ordinal value is preferred for ambiguous functions?
+                functions.put(functionName, d);
```

Comment:
Changing this so that the descriptor with the highest ordinal value is used in the case of ambiguity would be a breaking change for any pipelines that currently happen to be using the descriptor with the lowest ordinal value successfully. However, the only ambiguity I know of involves two descriptors whose ordinal is 0, and I am not sure if the loading order is even deterministic in that case. If not, then it seems like now would be as good a time as any to pin down the behavior.

Reply:
Yeah, I'm 100% fine with that.